sympy/solvers/tests/test_ode.py

```diff
@@ -3115,3 +3115,8 @@ def test_factoring_ode():
     soln = Eq(f(x), (C1*x**2/2 + C2*x + C3 - x)/(1 + x))
     assert checkodesol(eqn, soln, order=2, solve_for_func=False)[0]
     assert soln == dsolve(eqn, f(x))
+
+def test_issue_15913():
+    eq = -C1/x - 2*x*f(x) - f(x) + Derivative(f(x), x)
+    sol = C2*exp(x**2 + x) + exp(x**2 + x)*Integral(C1*exp(-x**2 - x)/x, x)
```

Comment:
I believe the issue would be solved if you correct the code that returned the integral to return the constant on the *outside* of the integral. Because it is on the inside, terms like `x*Integral(C1*x, x)` are not cancelling with terms like `-x*C1*Integral(x, x)`. The `doit` just takes the expensive route and computes the integrals. As was pointed out, this is not the economical way to proceed.

Reply:
This example comes indirectly from my (unpublished) attempts to add tests to the ODE module. We could change `dsolve` do it doesn't return this. However although `checkodesol` is primarily used for the ODE tests it is also a user facing function so I think it should be improved to handle this case.

Reply:
The use of doit on expressions that contain potentially intractable or slow integrals should not be part of the solution. Every case where a user wanted to simplify an expression containing an Integral could potentially be helped if constants that were not part of the variable of integration were moved to the outside of the Integral as they are for Sum. Compare `Sum(x*y, (x, 1, n)).simplify()` and `Integral(x*y, (x, 1, n)).simplify()`.  cf #15965